### ISSUE TITLE
to_number suporting float numbers

### ISSUE THREAD

**Issue by XRay3D** [hanickadot/compile-time-regular-expressions#298]
```std::from_chars``` for int is base defaulted as 10, you can skip it if nothing else is specified.
For floating point numbers, different arguments are used.
Using:
```cpp
value.to_number<double>(std::chars_format::general); // from "123.456"
value.to_number<double>(std::chars_format::hex); // from hex
value.to_number(); // from "123"
value.to_number(16); // from hex
```

https://github.com/hanickadot/compile-time-regular-expressions/blob/78ba3a3ddc6a977b0a46ad566876a6844eedb16a/single-header/ctre.hpp#L3636-L3638
Possible implementation
```cpp
template <typename R = int> 
constexpr CTRE_FORCE_INLINE auto to_number(auto... args) const noexcept -> R {
    return _captures.template select<0>().template to_number<R>(args...);
}
```

https://github.com/hanickadot/compile-time-regular-expressions/blob/78ba3a3ddc6a977b0a46ad566876a6844eedb16a/single-header/ctre.hpp#L3406-L3411
Possible implementation
```cpp
template <typename R = int> 
constexpr CTRE_FORCE_INLINE auto to_number(auto... args) const noexcept -> R {
    R result{};
    const auto view = to_view();
    std::from_chars(view.data(), view.data() + view.size(), result, args...);
    return result;
}
```

**Comment by hanickadot** [comment]
Good idea! Implemented in 029f1f13646cf65ec09780013418cb8f1c5d3a59.

**Comment by XRay3D** [comment]
Thank you.
